core/includes/common.inc

```diff
@@ -3796,6 +3823,9 @@ function backdrop_pre_render_styles($elements) {
 
   // Loop through each group.
   foreach ($elements['#groups'] as $group) {
+    // Ensure the group's attributes setting is an array before attempting to
+    // merge with the element's attributes.
+    $group['attributes'] = isset($group['attributes']) ? $group['attributes'] : array();
```

Comment:
It's not clear to me how `$group['attributes']` would have been set prior to this function. It's a little odd how this set of attributes gets added to every file within the group (in addition to the attributes set on the file itself). What is the way in which you would expect this to be used?

Reply:
That's a very good question @quicksketch! It's a bit of a waste of your time with my picking things out of a d.org patch without knowing what it all does. I'm not sure what it does. And it's not present in the backdrop_add_js function.

I had assumed that it was meant to combine the attributes of files that could be aggregated but then I had also added a line to exclude preprocessing if attributes is set (that part wasn't in the original patch). I might just try remove it and see.

Reply:
Actually I wonder if it needs to be set in `backdrop_group_css()`. I'm going to try that.

Clearly the tests need to be expanded to test these variations.

Reply:
I'll remove all the `$group['attributes']` functionality since it doesn't seem be worth it.